### ISSUE TITLE
how can I prevent onFirstSuccess process from terminating on fist changed file? 

### ISSUE THREAD

**Issue by amir-arad** [gilamran/tsc-watch#20]
I'm using tsc-watch like so: 
`tsc-watch --onFirstSuccess "webpack-dev-server"`

I'd like to keep the dev server running, as it monitors the output of tsc-watch.

**Comment by gilamran** [comment]
I wouldn't recommend using `tsc-wach` along with webpack. `tsc-watch` was intended for cases where you want to restart something, like your nodejs server or a test suite.

Webpack is able to listen to new files and rebuild using typescript (I'm using [awesome-typescript-loader](https://github.com/s-panferov/awesome-typescript-loader)). Using Webpack and an external compiler might cause webpack to refresh several times or even "miss" some changes, as compilation might output several files...

Do you have other reasons to "run-once" after a successful compilation?

**Comment by amir-arad** [comment]
generally speaking, chaining hot-reloading services (while avoiding the initial reload after first compilation) is the use case.

specifically, I find webpack's ts loaders lacking in some aspects and would like to manage the build chain myself. benefits are:
1. transpilation cache in plain FS format, ready for publish
2. multi-process dev server, taking advantage of more cores for cpu-heavy chires such as transpilation and bundling
3. simpler tools, less edge-case bugs (for instance, I don't use [hard-source-webpack-plugin](https://github.com/mzgoddard/hard-source-webpack-plugin) or [happypack](https://github.com/amireh/happypack))

**Comment by gilamran** [comment]
In your case (Specific use case), I would recommend using the library an not the command line tool.
This way, you can listen to events, and run your own "custom" code:
```
const watch = require('tsc-watch/client');
watch.on('first_success', () => {
  watch.kill();
});

```
WDYT?

**Comment by amir-arad** [comment]
that's the way to get it done with the current CLI, sure.
but what about a `onFirstSuccessTillExit` param?

**Comment by gilamran** [comment]
I don't **think** that this is a public need, If we get some +1s on that I will consider it. But as it is possible with the current implementation (Without adding new feature) I'd rather keep it simple for the general public.

You can always fork 😉

**Comment by gilamran** [comment]
Gr8, I see that you forked and implemented it on your own version.

**Comment by gilamran** [comment]
I'm happy to say that as of version 2.0.0 the `onFirstSuccess` will not get killed.

**Comment by iagomelanias** [comment]
The version 2.0.0 is not published on NPM. Was it intentional?

**Comment by gilamran** [comment]
@iagomelanias , you are right.
I just published version 2.1.2 as the latest.
